src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -10,9 +13,50 @@ package dsls.regex
 /** The top of a class hierarchy that encodes regular expressions. */
 abstract class RegularExpression {
   /** returns true if the given string matches this regular expression */
+  
   def matches(string: String) = RegexMatcher.matches(string, this)
+  
+  // || represents the Union of this, which is of type RegularExpression, 
+  // with "other", which is also of type RegularExpression
+  def || (other: RegularExpression) = Union(this, other)
```

Comment:
In Scala, for public methods, I think it is preferred to use the `/** */` style comments (like JavaDoc) instead of the `//` style comment when describing a method.